src/lib/Repository/DecoratedLocationService.php

```diff
@@ -66,14 +76,15 @@ function (Location $location) use ($locationService, $pathService, $contentInfo)
             },
             $locations
         );
+    }
 
+    private function unshiftMainLocation(array &$decoratedLocations)
```

Comment:
I'd go for a more semantic name. "unshift" is not _what_ the method does, it's _how_ it does it. What about "prioritizeMainLocation()", or something in this range ?

For consistency, I'd avoid the reference argument, and would make it return the newly organized array.